app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,58 @@
+<head>
+  <title>Recipes</title>
+  <%= stylesheet_link_tag 'application', 'recipes_show' %>
+</head>
+>
+<div class="recipe-details">
+  <h2 class="center-text"><%= @recipe.name %></h2>
+
+  <div class="recipe-info">
+    <div>
+      <p><strong>Preparation Time:</strong> <%= @recipe.preparation_time %></p>
+      <p><strong>Cooking Time:</strong> <%= @recipe.cooking_time %></p>
+      <p><%= @recipe.description %></p>
+    </div>
+    <div class="btn-container">
+      <% if @recipe.user == current_user %>
+        <%= form_with(model: @recipe, url: toggle_public_recipe_path(@recipe), method: :patch) do |form| %>
+        <div class="cbox">
+          <%= form.check_box :public, class: 'box' %>
+          <%= form.label :public, 'Make Recipe Public' %>
+          </div>
+          <%= form.submit 'Update Visibility' %>
+        <% end %>
+      <% end %>
+    </div>
+  </div>  
+
+  <div class="recipe-actions">
+    <%= link_to 'Generate Shopping List', generate_shopping_list_recipe_path(@recipe), class: "button" %>
+    <%= link_to "Add Ingredient", new_recipe_recipe_food_path(@recipe), class: "button" %>
+  </div>
+</div>
+
+<hr>
+
+<table>
+  <thead>
+    <tr>
+      <th>Food</th>
+      <th>Quantity</th>
+      <th>Value</th>
+      <th>Actions</th>
+    </tr>
+  </thead>
+  <tbody>
+    <% @recipe.recipe_foods.each do |recipe_food| %>
+      <tr>
+        <td><%= recipe_food.food.name %></td>
+        <td><%= recipe_food.quantity %></td>
+        <td><%= number_to_currency(recipe_food.food.price * recipe_food.quantity) %></td>
+        <td>
+          <%= link_to "Modify", edit_recipe_recipe_food_path(@recipe, recipe_food) %> 
+          <%= link_to "Remove", recipe_recipe_food_path(@recipe, recipe_food), method: :delete, data: { confirm: 'Are you sure?' } %>
+        </td>
```

Comment:
- [ ] I was unable to delete a recipe on the recipe show_page, I got the error message displayed below 👇 

<img width="422" alt="Untitled" src="https://github.com/lerfast/recipe-app/assets/103386183/45cb7a73-6dc2-4e6b-a361-3e807b649322">

Reply:
In my repo, that bottom works fine, however, I changed the code for that button:
![image](https://github.com/lerfast/recipe-app/assets/111823606/11af3ad8-2cdc-48aa-9228-fc0e5c103a4c)
![image](https://github.com/lerfast/recipe-app/assets/111823606/9f91c5e3-7aa6-4b91-a790-3ffbd7b57fcf)
![image](https://github.com/lerfast/recipe-app/assets/111823606/15ea2239-df90-4a73-a39a-a08cb7ac305e)
![image](https://github.com/lerfast/recipe-app/assets/111823606/0289cdb3-60a8-4ffb-b3e5-970f230c3300)
So I hope that this will work for you

Reply:
- Well resolved